packages/brand-shape/src/utils.ts

```diff
@@ -0,0 +1,368 @@
+      isLargeDot(...adjacentCoordinates['right']) &&
+      isLargeDot(...adjacentCoordinates['topRight']) &&
+      isLargeDot(...adjacentCoordinates['top'])
+    ) {
+      /**
+       *    ◜◝
+       * -> ◟◞
+       */
+      return Direction.BottomLeft;
+    }
+  }
+
+  /**
+   * Returns the coordinates for the quarter circle that shares the given face
+   * i.e. given `top`, returns coordinates of either ◜ or ◝
+   * Returns the given coordinates if this is not a large circle
+   */
+  function getLargeCircleSiblingIndex(
+    face: Direction,
+    [row, col]: Coordinate,
+  ): Coordinate {
+    const adjacentCoordinates = adjacentRowColumnCoordinates([row, col]);
+
+    if (isLargeDot(row, col)) {
+      switch (face) {
+        case Direction.Top:
+        case Direction.Bottom: {
+          // Is the index to the left part of the large circle?
+          if (isLargeDot(...adjacentCoordinates['left'])) {
+            return adjacentCoordinates['left'];
+          }
+
+          // index to the right?
+          if (isLargeDot(...adjacentCoordinates['right'])) {
+            return adjacentCoordinates['right'];
+          }
+
+          // Code error
+          return [row, col];
+        }
+
+        case Direction.Right:
+        case Direction.Left: {
+          // Is the index to the top part of the large circle?
+          if (isLargeDot(...adjacentCoordinates['top'])) {
+            return adjacentCoordinates['top'];
+          }
+
+          // index to the bottom?
+          if (isLargeDot(...adjacentCoordinates['bottom'])) {
+            return adjacentCoordinates['bottom'];
+          }
+
+          // Code error
+          return [row, col];
+        }
+      }
+    }
+
+    // Not in large circle
+    return [row, col];
+  }
+
+  return {
+    circleCount,
+    hasAdjacency,
+    indexExists,
+    isEmpty,
+    isNotEmpty,
+    isSmallDot,
+    isLargeDot,
+    findStart,
+    getLargeCircleSiblingIndex,
+    indexHasDiagonalAdjacencies,
+  };
+};
+
+/**
+ * Generates a record of the (row,column) pairs for each adjacent direction.
+ */
+export const adjacentRowColumnCoordinates = ([row, col]: Coordinate): Record<
+  Direction,
+  Coordinate
+> => {
+  return {
+    top: [row - 1, col],
+    topRight: [row - 1, col + 1],
+    right: [row, col + 1],
+    bottomRight: [row + 1, col + 1],
+    bottom: [row + 1, col],
+    bottomLeft: [row + 1, col - 1],
+    left: [row, col - 1],
+    topLeft: [row - 1, col - 1],
+  };
+};
+
+/**
+ * Generates a record of all the vertexes of a small circle given a center point
+ */
+export const vertexCoordinatesForCenterPoint = (
+  [x, y]: Coordinate,
+  size: CircleSize = 'small',
+): Record<CardinalDirection, Coordinate> => {
+  if (size === 'large') {
+    return {
+      top: [x, y - 1],
+      right: [x + 1, y],
+      bottom: [x, y + 1],
+      left: [x - 1, y],
+    };
+  }
+
+  return {
+    top: [x, y - 1],
+    right: [x + 1, y],
+    bottom: [x, y + 1],
+    left: [x - 1, y],
+  };
+};
```

Comment:
Using `(x,y)` values for the directions would probably help make these functions shorter too.